Add `ladybird.rb` formula

Neved4 opened the pull request.

0risc reviewed (changes requested):
As far as I can tell Ladybird doesn't provide any precompiled builds yet so this would have to be a formula rather than a cask. Spent a few minutes on it but it looks like it'll be fairly complicated to build.

Worth considering whether it's appropriate for a brew formula at this time https://github.com/LadybirdBrowser/ladybird/issues/25#issuecomment-2150211984

Also seems like you've got some extraneous commits in here.

Please do the following:
1. Move it to the formulas folder
1. Fix the style issues (tabs instead of spaces, wrong indentation)
1. Make it actually build (seemingly will need more dependencies and possibly more build commands. not 100% sure what changes will be required. I messed with it for a while and couldn't get it to build)
1. You may want to get rid of the extraneous commits that change `README.md` in unrelated ways

Neved4 commented:
Homebrew's [Cask DSL](https://docs.brew.sh/Cask-Cookbook) currently lacks the idioms for handling these operations cleanly and user-friendly. Closing this issue until a cleaner solution is available.

Neved4 commented:
@0risc
- [x] Fixed most `brew style` issues
- [x] Solved `README.md` conflicts

Neved4 commented:
We'd appreciate any outside help to build Ladybird 🤍

Neved4 commented:
For reference, the [Nix package](https://github.com/NixOS/nixpkgs/blob/7b254600a3121d0845b6b3b40816a7e5c2e7143d/pkgs/applications/networking/browsers/ladybird/default.nix) contains some build steps.

Neved4 commented:
There's been some progress and got rid of some of the errors when building it

For reference, it currently errors at:
```
CMake Error at /opt/homebrew/Cellar/cmake/3.30.4/share/cmake/Modules/FindPkgConfig.cmake:645 (message):
  The following required packages were not found:

   - skia=129
   - skia
```

Ladybird uses [Skia](https://github.com/google/skia) for rendering:
> **Skia now powers SVG and 2D canvas**
>
> _Last month, we adopted the third-party library Skia as our default rasterizer for high-performance 2D graphics. This month, we’ve expanded Skia’s role beyond HTML document rendering. It now powers the <canvas> element and SVG content, and also improves the speed and accuracy of CSS border rendering._
>
> Src: https://ladybird.org/newsletter/2024-08-31/#skia-now-powers-svg-and-2d-canvas

Currently [google/skia](https://github.com/google/skia) is not packaged by Homebrew, so we'll need to package it to build Ladybird.

Neved4 commented:
Decided to take a different approach and build it with the steps from [BuildInstructionsLadybird.md](https://github.com/LadybirdBrowser/ladybird/blob/master/Documentation/BuildInstructionsLadybird.md), namely:
```sh
brew install autoconf autoconf-archive automake ccache cmake ffmpeg nasm ninja pkg-config llvm

CC=$(brew --prefix llvm)/bin/clang CXX=$(brew --prefix llvm)/bin/clang++ ./Meta/ladybird.sh build
```

Neved4 commented:
Ladybird seems to be building, but we need to move the resulting `.app` to `/Applications`.

tostr7191 commented:
I installed the current `ladybird.rb`, and afterwards I get this:

```
$ ladybird 
dyld[39746]: Library not loaded: @rpath/liblagom-webview.0.dylib
  Referenced from: <7D37594E-D72D-3B29-81F9-636CCB53232B> /opt/homebrew/Cellar/ladybird/0.1.0-b9e7c6a/Ladybird.app/Contents/MacOS/Ladybird
  Reason: tried: '/opt/homebrew/Cellar/ladybird/0.1.0-b9e7c6a/Ladybird.app/Contents/lib/liblagom-webview.0.dylib' (no such file), '/opt/homebrew/Cellar/ladybird/0.1.0-b9e7c6a/Ladybird.app/Contents/lib/liblagom-webview.0.dylib' (no such file)
fish: Job 1, 'ladybird' terminated by signal SIGABRT (Abort)
```

The directory `/opt/homebrew/Cellar/ladybird/0.1.0-b9e7c6a/Ladybird.app/Contents/lib/` does, in fact, not exist. The buildscript did finish without any reported issues however.

Neved4 commented:
Hey @tostr7191 thx for trying this out!

Wasn't advancing with it personally, so I assigned it to @0risc.

We'd love any help to push this forward!

tostr7191 commented:
@Neved4 thanks for you words, I am not really knowledgeable about this subject unfortunately. But I did try to run the steps in the formula by hand, and - lo and behold - I get a functioning .app file in the build directory.

Upon further inspection, the problem seems to come from the `lib` in the app bundle being symlinked, but getting cleaned up by brew, which breaks the link.

I have changed the formula a bit, an now it builds a functioning .app bundle on my system, I would be grateful if someone could verify it. The main change is ln29-32, which replaces the `lib` symlink with the actual directory.

Formula follows, if it works I can create a proper pull request.

```ruby
class Ladybird < Formula
  desc "Truly independent web browser (pre-alpha developer version)"
  homepage "https://ladybird.org/"
  url "https://github.com/LadybirdBrowser/ladybird/archive/b9e7c6a2f66d83c6af73bcc102c05d3056382c02.tar.gz"
  version "0.1.0-b9e7c6a"
  sha256 "855a293849805d89c49cc29fd87a6b3e002f68d633f7ce52065e3260046be54b"
  license "BSD-2-Clause"
  head "https://github.com/LadybirdBrowser/ladybird.git", branch: "master"

  depends_on "autoconf" => :build
  depends_on "autoconf-archive" => :build
  depends_on "automake" => :build
  depends_on "ccache" => :build
  depends_on "cmake" => :build
  depends_on "ffmpeg" => :build
  depends_on "llvm" => :build
  depends_on "nasm" => :build
  depends_on "ninja" => :build
  depends_on "pkg-config" => :build
  depends_on :macos

  def install
    ENV["CC"] = "#{Formula["llvm"].opt_bin}/clang"
    ENV["CXX"] = "#{Formula["llvm"].opt_bin}/clang++"

    system "git", "init"
    system "./Meta/ladybird.sh", "install"

    cd "Build/ladybird/bin/Ladybird.app/Contents" do
      system "unlink", "./lib"
      system "cp", "-r", "../../../lib", "."
    end

    app_name = "Ladybird.app"
    app_path = "./Build/ladybird/bin/#{app_name}"

    prefix.install app_path
    bin.install_symlink "#{prefix}/#{app_name}/Contents/MacOS/Ladybird" => "ladybird"
    #bin.install_symlink "#{prefix}/#{app_name}" => "/Applications/"
  end

  def caveats
    <<~EOS
      Ladybird is in pre-alpha stage and is intended for developers only.
    EOS
  end
end
```

Neved4 commented:
@tostr7191 Woa thx a lot 🚀 LGTM! Added you as a Co-Author in the commit!

Also changed those `system` calls to more idiomatic ones to placate `brew style`.

Neved4 commented:
@tostr7191 Apparently some runners are failing, lmk if u can take a look: [?pr=40#step:7:55](https://github.com/Neved4/homebrew-tap/actions/runs/11871832046/job/33084828905?pr=40#step:7:55)

ADKaster commented:
Using vcpkg inside a brew formula is strange. I would expect that upstream would only accept a formula if you use other homebrew packages to obtain the dependencies.

Neved4 commented:
Hey @ADKaster thx I think you're right!

Do you have any improvements we could add?

Cheers
